Bootcamp - Python para dados/aula02/README.md

```diff
@@ -426,15 +426,16 @@ except ValueError:
 ### Exercício 22: Verificador de Palíndromo
 
 ```python
-entrada = input("Digite uma palavra ou frase: ")
-if isinstance(entrada, str):
-    formatado = entrada.replace(" ", "").lower()
-    if formatado == formatado[::-1]:
-        print("É um palíndromo.")
-    else:
-        print("Não é um palíndromo.")
+palavra = input('Forneça uma palavra ou frase: ')
+if palavra.isdigit():
+    print("Digite uma palavra ou frase, por favor")

```

Comment:
Ótima sugestão, @gecofever! O código está funcional, mas gostaria de propor alguns ajustes para torná-lo mais robusto e alinhado a boas práticas.

1. Uso de `isalpha()` no lugar de `isdigit()`:

- O uso de `isalpha()` é mais adequado para este caso, pois verifica se a entrada contém apenas caracteres alfabéticos (incluindo letras com acento, que são permitidas no português). O `isdigit()` valida apenas números e, por isso, não reflete o objetivo do exercício.

2. Organização da lógica:

- Ao estruturar o código, priorizamos a lógica principal no fluxo superior (fora do `else`), deixando o código mais claro e direto. Verificações negativas (`not`) podem ser usadas para validar entradas inválidas e retornar rapidamente, simplificando o fluxo.

3. Validação e clareza:

- Adicionei um tratamento explícito para entradas inválidas com `raise ValueError`, alinhando o código a boas práticas de validação de dados. Isso melhora a comunicação de erros e evita comportamentos inesperados.


Segue a sugestão com base nesses pontos:

```suggestion
try:
    entrada = input("Digite uma palavra ou frase: ")

    # Remove espaços e converte para minúsculas
    formatado = entrada.replace(" ", "").lower()

    # Verifica se há números ou símbolos na entrada
    if not formatado.isalpha():
        raise ValueError("A entrada deve conter apenas letras (acentos são permitidos).")
    
    # Verifica se é um palíndromo
    if formatado == formatado[::-1]:
        print(f'"{entrada}" é um palíndromo.')
    else:
        print(f'"{entrada}" não é um palíndromo.')

except ValueError as ve:
    print(f"Erro: {ve}")
except Exception as e:
    print(f"Erro inesperado: {e}.")
```